src/unix/bsd-ifaddrs.c

```diff
@@ -42,12 +47,13 @@ static int uv__ifaddr_exclude(struct ifaddrs *ent) {
    * devices.  We're not interested in this information.
    */
   if (ent->ifa_addr->sa_family == AF_LINK)
-    return 1;
+    return !!(int)type;
+  return !(int)type;
 #elif defined(__NetBSD__) || defined(__OpenBSD__)
-  if (ent->ifa_addr->sa_family != PF_INET)
+  if (type == UV__EXCLUDE_IFADDR && ent->ifa_addr->sa_family != PF_INET)
     return 1;
```

Comment:
You can drop the check and just `return !exclude_type;` or `return exclude_type == UV__EXCLUDE_IFADDR;`.